### PR TITLE
Hide the date-clear icon and style vaadin-date-input like a paper input

### PR DESCRIPTION
**Pull request by NovapaX**
Hiding the 'clear' button as date input is required and clearing it throws a javascript error.

Used the theming methode showed here: https://cdn.vaadin.com/vaadin-date-picker/2.0.2/demo/theming.html

**screenshots of the new styling**
not focused:
schermafbeelding 2018-02-28 om 12 19 30

focused:
schermafbeelding 2018-02-28 om 12 21 29

### CODE REVIEW

File: src/resources/ha-date-picker-style.html [hunk 1]
```diff
@@ -0,0 +1,93 @@
+<dom-module id="ha-date-picker-vaadin-date-picker-styles" theme-for="vaadin-date-picker">
+  <template>
+    <style>
+      [part~="clear-button"] {
```
Comment (balloob) [inline src/resources/ha-date-picker-style.html]: This disables the clear button on all vaadin-date-pickers. That's not good.

File: src/resources/ha-date-picker-style.html [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        display: none;
+      }
+
+      [part~="toggle-button"] {
+        color: var(--disabled-text-color);
```
Comment (balloob) [inline src/resources/ha-date-picker-style.html]: I think you want to use secondary-text-color ?

Comment (NovapaX) [reply src/resources/ha-date-picker-style.html]: I copied that var from for example the drop-down list arrow. But that makes more sense.

File: src/resources/ha-date-picker-style.html (hunk 1 above)
Comment (NovapaX) [reply src/resources/ha-date-picker-style.html]: True. I first targeted this to the history panel. Then realized those style template apply to all pickers once loaded, which is fine of course, except for hiding the clear button indeed.
I think I can make that depend on a ‘required’ attribute.

File: src/resources/ha-date-picker-style.html (hunk 2 above)
Comment (NovapaX) [reply src/resources/ha-date-picker-style.html]: I also see that button needs to move a couple of px to the left...

**Comment by NovapaX** [comment]
required is also an 'official' valid attribute on the date-picker. Doesn't seem to affect anything however.

**Review by balloob — APPROVED** [review]

**Pull request merged**